lib/archethic/contracts.ex

```diff
@@ -165,6 +167,41 @@ defmodule Archethic.Contracts do
     end
   end
 
+  @doc """
+  Simulate the execution of the contract hold in prev_tx with the inputs of next_tx, at a certain date
+  """
+  @spec validate_contract_execution(Transaction.t(), Transaction.t(), DateTime.t()) :: map()
+  def validate_contract_execution(
+        prev_tx = %Transaction{data: %TransactionData{code: code}},
+        next_tx = %Transaction{},
+        date = %DateTime{}
+      ) do
+    case Interpreter.parse(code) do
+      {:ok,
+       %Contract{
+         triggers: triggers,
+         conditions: %{transaction: transaction_conditions}
+       }} ->
+        constants = %{
+          "previous" => Constants.from_transaction(prev_tx),
+          "next" => Constants.from_transaction(next_tx)
+        }
+
+        with :ok <- validate_conditions(transaction_conditions, constants),
+             :ok <- validate_triggers(triggers, next_tx, date),
+             %Transaction{} <-
+               ActionInterpreter.execute(Map.fetch!(triggers, :transaction), constants) do
+          %{"valid" => true, "address" => prev_tx}
+        else
+          {:error, reason} ->
+            %{"valid" => false, "address" => prev_tx, "reason" => reason}
```

Comment:
May I suggest to not use the `%{"field" => value}` format?  (and use `%{field: value}` instead)
I find it really error prone and we should only deal with that format when we parse json user input.

Reply:
Ok 👍fixed.